eclipse-theia/theia
# Codicon color and URI support to TerminalOptions

This is a copy of the original PR #12861
First commit was cherry picked from https://github.com/FernandoAscencio/theia/commit/ba3dda24e7fac3ac81522d61d2ad531b4a94f8c9
In the second commit, I applied the suggested changed from the original PR. 

The original description below

#### What it does


Closes https://github.com/eclipse-theia/theia/issues/12074

This commit adds URI and { light, dark } support for terminal icons.
This commit adds ColorTheme support for ThemeIcons
Support for ExtensionTerminalOptions#iconPath

#### How to test

Open Theia
Install from VSIX [the test extension](https://github.com/eclipse-theia/theia/files/12434150/terminalcreationtests-0.0.1.zip)
Check the result follow VSCode test results seen below:


VSCode test Results
Dark
<img width="190" alt="TerminalIconTests" src="https://github.com/eclipse-theia/theia/assets/48699277/fe36b9c4-9ea8-450f-8d36-6dfa165f85ff">
Light
<img width="192" alt="TerminalIconTestsLight" src="https://github.com/eclipse-theia/theia/assets/48699277/ea7f19ca-f685-493e-b011-20723b88b5f7">

#### Review checklist

- [ x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
